compiler/rustc_mir_build/src/thir/pattern/deconstruct_pat.rs

```diff
@@ -630,9 +640,9 @@ pub(super) enum Constructor<'tcx> {
     /// Ranges of integer literal values (`2`, `2..=5` or `2..5`).
     IntRange(IntRange),
     /// Ranges of floating-point literal values (`2.0..=5.2`).
-    FloatRange(ty::Const<'tcx>, ty::Const<'tcx>, RangeEnd),
+    FloatRange(mir::ConstantKind<'tcx>, mir::ConstantKind<'tcx>, RangeEnd),
     /// String literals. Strings are not quite the same as `&[u8]` so we treat them separately.
-    Str(ty::Const<'tcx>),
+    Str(mir::ConstantKind<'tcx>),
```

Comment:
as we structurally match strings, we should use `ty::Const<'tcx>` here

Reply:
or tbh, probably we should use `Str(Valtree)` :sweat_smile: not sure what's the best way to do this

Reply:
`&'tcx [u8]` would be the least ambiguous, but may require some conversions or interning later. Right now you can get it right from the `Allocation`